internal/provider/data_source_http.go

```diff
@@ -162,12 +163,29 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 
 	caCertificate := model.CaCertificate
 
-	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{},
+	tr, ok := http.DefaultTransport.(*http.Transport)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Error configuring http transport",
+			"Error http: Can't configure http transport.",
+		)
+		return
+	}
+
+	// Prevent issues with tests modifying shared transport.
+	clonedTr := tr.Clone()
+
+	// Prevent issues with tests caching the proxy configuration.
+	clonedTr.Proxy = func(req *http.Request) (*url.URL, error) {
+		return httpproxy.FromEnvironment().ProxyFunc()(req.URL)
+	}
+
+	if clonedTr.TLSClientConfig == nil {
+		clonedTr.TLSClientConfig = &tls.Config{}
 	}
 
 	if !model.Insecure.IsNull() {
-		tr.TLSClientConfig.InsecureSkipVerify = model.Insecure.ValueBool()
+		clonedTr.TLSClientConfig.InsecureSkipVerify = model.Insecure.ValueBool()
```

Comment:
Refer to above comment

```suggestion
		if clonedTr.TLSClientConfig == nil {
			clonedTr.TLSClientConfig = &tls.Config{}
		}
		clonedTr.TLSClientConfig.InsecureSkipVerify = model.Insecure.ValueBool()
```